### ISSUE TITLE
Metals gives up printing to the debug console when System.out is spammed

### ISSUE THREAD

**Issue by MRandl** [scalameta/metals#2043]
**Describe the bug**

When running the program described below, which is basically made to spam the sysout flow, metals will at a random point give up reporting the prints on the 'debug console' tab. However, the program itself executes correctly : the outputs are correctly logged in the 'output' tab, albeit in a less readable fashion. The JVM exits gracefully when expected to.

**To Reproduce**
1. Generate a new dotty project (lampepfl/dotty.g8). This bug was not tested on mainstream 2.x scala.
2. Delete the src/test folder, open the main source file and copy the following code :
```scala
object Main {

  def main(args: Array[String]): Unit = {
    val stringList = List.range(0, 26).map(c => (c + 'A').toChar.toString)
    val stringCombo = 
      for {
        i <- Seq.range(0, stringList.size)
        j <- stringList.drop(i)
      } yield {
        s"{${stringList(i)}, $j} ~> ${stringEquals(stringList(i), j)}"
      }

    stringCombo.foreach(println)
    System.out.flush
  }

  def stringEquals(a : String, b : String) : Boolean =
    if(a.isEmpty || b.isEmpty)
      a == b
    else
      a.head == b.head && stringEquals(a.tail, b.tail) 


}
```
3. Run until the bug appears. This may take a few attempts.

The program generates every combination of two letters and prints them, along with a boolean equal to true iff the letters are the same.

**Expected behavior**

We should see all combinations of letters appear on the debug console. This is sometimes not the case.

**Screenshots**

![Capture d’écran de 2020-09-01 14-34-31](https://user-images.githubusercontent.com/17947709/91851953-9388a980-ec4f-11ea-8498-c8913086111c.png)
It should stop after comparing Z and Z. Here, it stops at J and P.

**Installation:**
- Operating system: tested on windows and ubuntu, both up to date
- Editor: Visual Studio Code
- Metals version: v0.9.3
- JVM version : openjdk 1.8 but I don't think this is relevant

**Comment by kpbochenek** [comment]
Thanks for reporting!

Also it reproduces very easy on given example :+1:

Looks like something forwarding output to vscode stops before all output is flushed that's why all output doesn't reach vscode console.

Just some things quickly tested:
sbt -> always produces all output on console
bloop -> always produces all output on console
from vscode -> always output is truncated
from vscode (adding Thread.sleep(1000) in the end) -> output produced correctly.

**Comment by MRandl** [comment]
That means the stream bringing output data from the program to the console dies with the JVM? That explains it I guess... Also that means the problem is unrelated to spamming the flow, really

**Comment by kpbochenek** [comment]
yes exactly, too early termination of output proxy in metals is most likely the issue.
